src/rules/noWarningCommentsRule.ts

```diff
@@ -0,0 +1,175 @@
+
+    public static COMMENT_ERROR_FACTORY = (term: string, location: "start" | "anywhere"): string =>
+        `Warning term '${term}' found ${location === "start" ? "at start of" : "in"} comment`
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithFunction(sourceFile, walk, parseOptions(this.ruleArguments[0] as IOptionsObject));
+    }
+}
+
+function parseOptions(options: IOptionsObject): Options {
+    if (typeof options === "undefined" || options === null) {
+        options = {};
+    }
+
+    if (
+        (typeof options.location !== "undefined" && options.location !== null) &&
+        options.location !== "start" &&
+        options.location !== "anywhere") {
+        throw new Error(`Option 'location' value is invalid: ${options.location}`);
+    }
+
+    if (
+        (typeof options.terms !== "undefined" && options.terms !== null) &&
+        !(options.terms instanceof Array)) {
+        throw new Error("Option 'terms' must be a string array.");
+    }
+
+    return {
+        location: (typeof options.location !== "undefined" && options.location !== null) ?
+            options.location as "start" | "anywhere" :
+            "start",
+        terms: (typeof options.terms !== "undefined" && options.terms !== null) ?
+            options.terms :
+            ["todo", "fixme", "xxx"],
+    };
+}
+
+function checkCommentContainsWarningTerms(comment: string, regExps: RegExp[], rng: ts.CommentRange, ctx: Lint.WalkContext<Options>) {
+    regExps.forEach((regExp) => {
+        const match = regExp.exec(comment);
+        if (typeof match !== "undefined" && match !== null) {
+            ctx.addFailure(
+                rng.pos + match[1].length,
+                rng.pos + match[1].length + match[2].length,
+                Rule.COMMENT_ERROR_FACTORY(match[2], ctx.options.location!));
+        }
+    });
+}
+
+function convertToRegExp(term: string, location: "start" | "anywhere") {
+    const escaped = utils.escapeRegExp(term);
+    let prefix;
+
+    /*
+     * If the term ends in a word character (a-z0-9_), ensure a word
+     * boundary at the end, so that substrings do not get falsely
+     * matched. eg "todo" in a string such as "mastodon".
+     * If the term ends in a non-word character, then \b won't match on
+     * the boundary to the next non-word character, which would likely
+     * be a space. For example `/\bFIX!\b/.test('FIX! blah') === false`.
+     * In these cases, use no bounding match. Same applies for the
+     * prefix, handled below.
+     */
+    const suffix = /\w$/.test(term) ? "\\b" : "";
+
+    if (location === "start") {
+        /*
+         * When matching at the start, ignore leading whitespace, and
+         * there's no need to worry about word boundaries.
+         */
+        prefix = "^[\\s\\/\\*]*";
+    } else if (/^\w/.test(term)) {
+        prefix = "^[\\s\\S]*?\\b";
+    } else {
+        prefix = "^[\\s\\S]*?";
```

Comment:
please add a comment that `[\s\S]` is used to match every character including line breaks